src/playercards/PlayerCardPanel.ttslua

```diff
@@ -0,0 +1,509 @@
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L0 Mystic", click_function="spawnBasicMystic", function_owner=self,
+      position={-0.15,0.2,0.5}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L1-5 Mystic", click_function="spawnUpgradedGuardian", function_owner=self,
+      position={0.15,0.2,0.5}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L0 Rogue", click_function="spawnBasicRogue", function_owner=self,
+      position={-0.15,0.2,0.6}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L1-5 Rogue", click_function="spawnUpgradedRogue", function_owner=self,
+      position={0.15,0.2,0.6}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L0 Survivor", click_function="spawnBasicSurvivor", function_owner=self,
+      position={-0.15,0.2,0.7}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L1-5 Survivor", click_function="spawnUpgradedSurvivor", function_owner=self,
+      position={0.15,0.2,0.7}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L0 Neutral", click_function="spawnBasicNeutral", function_owner=self,
+      position={-0.15,0.2,0.8}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="L1-5 Neutral", click_function="spawnUpgradedNeutral", function_owner=self,
+      position={0.15,0.2,0.8}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="Clear", click_function="deleteAll", function_owner=self,
+      position={0.5,0.2,0.9}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
+  self.createButton({
+      label="Weaknesses", click_function="spawnWeaknesses", function_owner=self,
+      position={-0.5,0.2,0.9}, rotation={0,0,0}, height=BUTTON_WIDTH, width=BUTTON_HEIGHT,
+      font_size=64, color={0,0,0}, font_color={1,1,1}, scale={0.25, 0.25, 0.25}
+  })
```

Comment:
The button creation begs for usage of a variable with common parameters and/or a loop to create these.
It would also be better to have a click_function with parameters instead of a separate function for each button (can be achieved by `setVar`)

Reply:
Absolutely.  But the button creation is all going to need to be rewritten once the new UI panel is added, so I didn't think it was worth it yet since it was all going to change anyway.

I do already loop the function creation for things that I knew the values for.  Release names I was going to let CZ define while adding the data rather than enforcing it on him ahead of time.